### PR TITLE
fix(files_sharing): Add proper user facing messages on success

### LINKED ISSUE #38157: [Bug]: Share confirmations messages wrong
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] Nextcloud Server **is** running on 64bit capable CPU, PHP and OS.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

https://github.com/nextcloud/server/blob/fc076271c791998272856b640be4b63cfe3263a2/apps/files_sharing/src/mixins/SharesMixin.js#L280

See screenshot

image

### Steps to reproduce

1. Create a share link
2. Update expiration date or Hide dowload
3. See confirmation messages

### Expected behavior

Variables are implemented (and translated)

### Nextcloud Server version

25

### PR DESCRIPTION
**Pull request by susnux**
* Resolves: https://github.com/nextcloud/server/issues/38157

## Summary

1. Use proper translated strings
2. Fix invalid use of translation message on API result
3. Log all errors

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [ ] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by Ivansss — APPROVED** [review]

File: apps/files_sharing/src/mixins/SharesMixin.js [hunk 1]
```diff
@@ -333,6 +339,32 @@ export default {
...
+		updateSuccessMessage(names) {
+			if (names.length !== 1) {
+				return t('files_sharing', 'Share saved')
+			}
+
+			switch (names[0]) {
+				case 'expireDate':
+					return t('files_sharing', 'Share expire date saved')
+				case 'hideDownload':
+					return t('files_sharing', 'Share hide-password state saved')
```
Comment (Pytal) [inline apps/files_sharing/src/mixins/SharesMixin.js:354]:
```suggestion
					return t('files_sharing', 'Share hide-download state saved')
```

**Review by Pytal — APPROVED** [review]
Just need to fix the message 👍

**Pull request merged**